src/components/Data/Collections/Watch.vue

```diff
@@ -427,111 +360,69 @@ export default {
       this.subscribed = false
       this.room = null
     },
-    onReset(newFilters) {
-      filterManager.saveToRouter(
-        filterManager.stripDefaultValuesFromFilter(newFilters),
-        this.$router
-      )
-      this.reset()
-    },
-    reset() {
-      // trigged when user changed the collection of watch data page
-      this.notifications = []
-      this.warning.message = ''
-      this.warning.count = 0
-
+    resetFilters() {
       if (this.subscribed) {
         this.subscribed = false
         this.unsubscribe(this.room)
       }
+      this.currentFilter.raw = {}
+      this.$refs.rawFilter.resetSearch()
     },
-    clear() {
-      this.warning.message = ''
+    resetNotifications() {
       this.notifications = []
+      this.warning.message = ''
+      this.warning.count = 0
     },
-    computeNotifHeight() {
-      Vue.nextTick(() => {
-        const mainNavHeight = document.getElementById('mainnav').offsetHeight
-        const searchFilter = document.getElementsByClassName('Filters')[0]
-          .offsetHeight
-        const subCtrl = this.$refs.subscribeControl.offsetHeight
-        const notifHeight =
-          document.body.offsetHeight - (mainNavHeight + searchFilter + subCtrl)
-
-        this.notifStyle = { maxHeight: notifHeight + 'px', overflowY: 'auto' }
-      })
+    reset() {
+      // trigged when user changed the collection of watch data page
+      this.resetFilters()
+      this.resetNotifications()
+    }
+  },
+  watch: {
+    index() {
+      this.reset()
     },
-    setScrollDown(v) {
-      this.scrollDown = v
+    collection() {
+      this.reset()
     },
-    handleWebNotification(text) {
-      const notif = new window.Notification('Kuzzle Admin Console', {
-        body: text + ' in ' + this.index + ' ' + this.collection,
-        icon: '/static/favicon/favicon-32x32.png'
-      })
-      setTimeout(notif.close.bind(notif), 5000)
+    $route() {
+      this.reset()
+      this.currentFilter = filterManager.loadFromRoute(this.$route)
+    }
+  },
+  async destroyed() {
+    this.reset()
+    if (this.room) {
+      await this.$kuzzle.realtime.unsubscribe(this.room)
     }
   }
 }
 </script>
 
 <style rel="stylesheet/scss" lang="scss">
```

Comment:
Why not scoped style here? 
if needed it is possible to put 2 styles tags, one scoped and the other not, which allows to better separate the styles and to be sure to have a minimum of css override